opensearch/src/main/java/org/opensearch/sql/opensearch/storage/OpenSearchIndexScan.java

```diff
@@ -152,9 +167,13 @@ public void pushDownSort(List<SortBuilder<?>> sortBuilders) {
 
   /**
    * Push down size (limit) and from (offset) to DSL request.
+   * Size is constrained by max result window size to allow for scrolling.
    */
   public void pushDownLimit(Integer limit, Integer offset) {
     SearchSourceBuilder sourceBuilder = request.getSourceBuilder();
+    if (limit + offset > maxResultWindow) {
+      limit = maxResultWindow - offset;
+    }
```

Comment:
This is for scrolling.
OpenSearch requires that from(offset) + size(limit) <= `index.max_result_window`. A limit operator with limit+offset > `index.max_result_window` will invoke scroll request multiple times, each with batch size `index.max_result_window`

Reply:
the if condition `limit + offset > maxResultWindow` is used in different place, could we simply it?